src/Controllers/LoginController.php

```diff
@@ -56,11 +57,23 @@ public function loginAction()
     }
 
     /**
-     * Logs the current user out.
+     * Returns the rules that should be used for validation.
      *
-     * @return Response|string
+     * @return string[]
+     */
+    protected function getValidationRules(): array
+    {
+        return setting('Validation.login') ?? [
+            //'username' => 'required|max_length[30]|alpha_numeric_space|min_length[3]',
+            'email'    => 'required|valid_email|max_length[254]',
```

Comment:
If too long value comes, it is better to invalidate immediately.

```suggestion
            'email'    => 'required|max_length[254]|valid_email',
```